### PR TITLE
[dom] Add Boost 1.75.0 with CrayPE 20.11

### PR DESCRIPTION
**Pull request by lucamar**
I provide the recipes and the patch for Boost 1.75.0 on Dom with Cray Programming Environment 20.11 (required by @aurianer  for HPX 1.6.0).

### CODE REVIEW

**Review by aurianer — APPROVED** [review]
LGTM thanks!

File: easybuild/easyconfigs/b/Boost/Boost-1.75.0-CrayGNU-20.11-python3.eb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+homepage = 'http://www.boost.org/'
+description = """
+    Boost provides free peer-reviewed portable C++ source libraries.
+"""
+
+toolchain = {'name': 'CrayGNU', 'version': '20.11'}
+toolchainopts = {'usempi': True, 'pic': True, 'verbose': False}
+
+source_urls = [SOURCEFORGE_SOURCE]
+sources = ['%(namelower)s_1_70_0.tar.gz']
```
Comment (msimberg) [inline easybuild/easyconfigs/b/Boost/Boost-1.75.0-CrayGNU-20.11-python3.eb:15]:
```suggestion
sources = ['%(namelower)s_1_75_0.tar.gz']
```
?

File: easybuild/easyconfigs/b/Boost/Boost-1.75.0-CrayGNU-20.11.eb [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+homepage = 'http://www.boost.org/'
+description = """
+    Boost provides free peer-reviewed portable C++ source libraries.
+"""
+
+toolchain = {'name': 'CrayGNU', 'version': '20.11'}
+toolchainopts = {'usempi': True, 'pic': True, 'verbose': False}
+
+source_urls = [SOURCEFORGE_SOURCE]
+sources = ['%(namelower)s_1_70_0.tar.gz']
```
Comment (msimberg) [inline easybuild/easyconfigs/b/Boost/Boost-1.75.0-CrayGNU-20.11.eb:14]:
```suggestion
sources = ['%(namelower)s_1_75_0.tar.gz']
```

File: easybuild/easyconfigs/b/Boost/Boost-1.75.0-python3.patch [hunk 3]
```diff
@@ -0,0 +1,23 @@
+diff -Naur boost_1_74_0.ori/tools/build/src/tools/python.jam boost_1_74_0/tools/build/src/tools/python.jam
+--- boost_1_74_0.ori/tools/build/src/tools/python.jam	2020-08-11 16:57:21.000000000 +0200
++++ boost_1_74_0/tools/build/src/tools/python.jam	2020-11-10 10:12:42.002540188 +0100
```
Comment (msimberg) [inline easybuild/easyconfigs/b/Boost/Boost-1.75.0-python3.patch:3]:
```suggestion
--- boost_1_75_0.ori/tools/build/src/tools/python.jam	2020-08-11 16:57:21.000000000 +0200
+++ boost_1_75_0/tools/build/src/tools/python.jam	2020-11-10 10:12:42.002540188 +0100
```

File: easybuild/easyconfigs/b/Boost/Boost-1.75.0-python3.patch [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+         includes ?= $(prefix)/include/python$(version) ;
++        includes = [ GLOB $(prefix)/include : * ] ;
++        includes = $(includes[1]) ;
++        ECHO "Using include path:" $(includes) ;
+ 
+         local lib = $(exec-prefix)/lib ;
+         libraries ?= $(lib)/python$(version)/config $(lib) ;
+diff -Naur boost_1_74_0.ori/user-config.jam boost_1_74_0/user-config.jam
+--- boost_1_74_0.ori/user-config.jam	1970-01-01 01:00:00.000000000 +0100
++++ boost_1_74_0/user-config.jam	2020-11-10 10:22:37.673600000 +0100
```
Comment (msimberg) [inline easybuild/easyconfigs/b/Boost/Boost-1.75.0-python3.patch:16]:
```suggestion
--- boost_1_75_0.ori/user-config.jam	1970-01-01 01:00:00.000000000 +0100
+++ boost_1_75_0/user-config.jam	2020-11-10 10:22:37.673600000 +0100
```

**Review by msimberg — COMMENTED** [review]
Thanks @lucamar for bumping this! Do the production lists also need to be updated for this to be visible?

File: easybuild/easyconfigs/b/Boost/Boost-1.75.0-CrayGNU-20.11.eb (hunk 2 above)
Comment (lucamar) [reply easybuild/easyconfigs/b/Boost/Boost-1.75.0-CrayGNU-20.11.eb]: Thanks, updated

File: easybuild/easyconfigs/b/Boost/Boost-1.75.0-CrayGNU-20.11-python3.eb (hunk 1 above)
Comment (lucamar) [reply easybuild/easyconfigs/b/Boost/Boost-1.75.0-CrayGNU-20.11-python3.eb]: Thanks, updated

**Comment by lucamar** [comment]
> Thanks @lucamar for bumping this! Do the production lists also need to be updated for this to be visible?

If you need the recipes as a dependency of HPX, we don't need to add them to the production list: they will be built as soon as HPX is built as dependencies and therefore will be visible as stand-alone modulefiles.

**Comment by msimberg** [comment]
> > Thanks @lucamar for bumping this! Do the production lists also need to be updated for this to be visible?
> 
> If you need the recipes as a dependency of HPX, we don't need to add them to the production list: they will be built as soon as HPX is built as dependencies and therefore will be visible as stand-alone modulefiles.

That'll be fine I think. We want them for the HPX recipe, but also for CI, so as long as it can be loaded that's good enough. If I remember correctly it can still be loaded with `Boost/.<version>` right?

**Review by msimberg — APPROVED** [review]
Thanks!

**Comment by lucamar** [comment]
> That'll be fine I think. We want them for the HPX recipe, but also for CI, so as long as it can be loaded that's good enough. If I remember correctly it can still be loaded with `Boost/.<version>` right?

No, it won't be built until it gets to the production list or a software that requires it as a dependency does.

**Comment by msimberg** [comment]
> > That'll be fine I think. We want them for the HPX recipe, but also for CI, so as long as it can be loaded that's good enough. If I remember correctly it can still be loaded with `Boost/.<version>` right?
> 
> No, it won't be built until it gets to the production list or a software that requires it as a dependency does.

Ok, we'll add the updated HPX recipe so we'll get it then latest.

We need python, but not the boost library. I think we depend on the python version of boost just because.

**Comment by lucamar** [comment]
> We need python, but not the boost library. I think we depend on the python version of boost just because.

Do you mean that you don't need `python` in the Boost library?

**Comment by msimberg** [comment]
> > We need python, but not the boost library. I think we depend on the python version of boost just because.
> 
> Do you mean that you don't need `python` in the Boost library?

Right, exactly. Sorry for making that unnecessarily unclear!

**Comment by lucamar** [comment]
> Right, exactly. Sorry for making that unnecessarily unclear!

Ok, if you approve, I'll merge and you'll be able to use the recipe without `python` as a dependency in your future pull requests. This pull request impacts Dom with PE 21.02: if you and @aurianer need it on Piz Daint with 20.08, I can open a different PR.

**Comment by msimberg** [comment]
> > Right, exactly. Sorry for making that unnecessarily unclear!
> 
> Ok, if you approve, I'll merge and you'll be able to use the recipe without `python` as a dependency in your future pull requests. This pull request impacts Dom with PE 20.11: if you and @aurianer need it on Piz Daint with 20.08, I can open a different PR.

Sounds good, thank you! We can take it from here for the Piz Daint and HPX recipes.

**Review by msimberg — APPROVED** [review]

**Pull request merged**